submissions/webdevtoday/oop_exercise/index.js

```diff
@@ -0,0 +1,80 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammal {
+   constructor(name, gender, saying, species, friends) {
+      this.name = name;
+      this.gender = gender;
+      this.species = species;
+      this.saying = saying;
+      this.friends = friends;
+      this.legs = 4;
+   }
+   getProperties() {
+      return [this.species, this.name, this.gender, this.legs, this.friends.join(', ')];
+   }
+   [Symbol.toPrimitive](hint) {
+      if (hint == 'string') {  
+         return this.getProperties().join(';');
+      }
+   }
+}
+
+class Human extends Mammal {
+   constructor(name, gender, saying, friends) {
+      super(name, gender, saying, 'human', friends);
+      this.legs = 2;
+      this.hands = 2;
+   }
+
+   getProperties() {
+      return [this.species, this.name, this.gender, this.legs, this.hands, this.friends.join(', ')];
+   }
+}
+
+class Animal extends Mammal {}
+
+const dog = new Animal('Jhonny', 'male', 'woof-woof!', 'dog', ['Andrew', 'Anastasia']);
+const cat = new Animal('Murka', 'female', 'meow!', 'cat', ['Anastasia']);
+const woman = new Human('Anastasia', 'female', 'I\'m a woman!', ['Andrew', 'Murka', 'Jhonny']);
+const man = new Human('Andrew', 'male', 'I\'m a man!', ['Anastasia', 'Jhonny', 'Murka']);
+const catwoman = new Human('CatWoman', 'female', cat.saying, []);
```

Comment:
`Animal` (or `Mammal` when you get rid of an "empty" `Animal`) is a generic class that requires a species being explicitly specified, while `Human` is more specific and implies a particular species. A reader is able to catch a human's species right from its name.

How about having `Cat` and `Dog` classes? This way an inhabitant will be created from the class where name is a species name. This way code would look more consistent.